File: NeetCode/69/69.cpp
class Solution {

private:
    vector<vector<int>> res;
    vector<int> curNum;

    void dfs(int i, vector<int>& nums, int target, int suma){
        if(suma > target || i >= nums.size()){
            return;
        }
        else if(suma == target){
            res.push_back(curNum);
            return;
        }
        curNum.push_back(nums.at(i));
        dfs(i, nums, target, suma + nums.at(i));
        curNum.pop_back();
        dfs(i + 1, nums, target, suma);
    }

public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        dfs(0, nums, target, 0);
        return res;
    }
};

File: NeetCode/69/69.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        curNum = []

        def dfs(i: int) -> None:
            suma = sum(curNum)
            if suma > target or i >= len(nums):
                return
            elif suma == target:
                res.append(curNum.copy())
                return
            curNum.append(nums[i])
            dfs(i)
            curNum.pop()
            dfs(i + 1)

        dfs(0)
        return res